custom_visibility/src/custom/data/visibility/FilterOpportunities.php

```diff
@@ -89,11 +99,20 @@ public function addVisibilityWhere(&$query)
     }
 
     /**
-     * {@inheritdoc}
+     * orvewrite parent method, SugarVisibility::addVisibilityWhereQuery
+     *
+     * add visibility, i.e. condition, clauses to the WHERE part of SugarQuery
+     *
+     * @param SugarQuery $query
+     * @param array $options
+     * @return SugarQuery
      */
     public function addVisibilityWhereQuery(SugarQuery $sugarQuery, $options = array()) {
+        // create where clause
         $where = null;
         $this->addVisibilityWhere($where, $options);
+
+        // add the where clause to $sugarQuery's where part
         if (!empty($where)) {
```

Comment:
What happens if the $where query is empty?

Reply:
I guess a better question is when would this be likely?

Reply:
if $where is empty, there is nothing to add to existing WHERE clause